app/controllers/activitypub/collections_controller.rb

```diff
@@ -4,12 +4,13 @@ class ActivityPub::CollectionsController < Api::BaseController
   include SignatureVerification
   include AccountOwnedConcern
 
+  before_action :require_signature!, if: :secure_mode_enabled?
   before_action :set_size
   before_action :set_statuses
   before_action :set_cache_headers
 
   def show
-    expires_in 3.minutes, public: true
+    expires_in 3.minutes, public: true unless secure_mode_enabled?
```

Comment:
`expires_in` can still be set if public is set to false because `private` means "no shared caches"

Reply:
Okay